src/libsync/propagatedownload.cpp

```diff
@@ -544,6 +545,13 @@ void PropagateDownloadFile::slotGetFinished()
     ASSERT(job);
 
     SyncFileItem::Status status = job->errorStatus();
+
+    // Needed because GETFileZsyncJob may emit finishedSignal without any further network activity
+    if (!job->reply() && status == SyncFileItem::FileIgnored) {
```

Comment:
Ah, I see. It's again a case where `reply()` is never empty for real AbstractNetworkJobs, but this peculiar agglomoration of jobs might have it 0. I'd even remove the check on `status` then, because a null reply would lead to crashes further down if it isn't caught here.

Reply:
It should never get here with no `reply()` unless `status == SyncFileItem::FileIgnored`, I didn't want to catch any other unexpected path here and `succeed` by mistake, I personally prefer a crash so a bug is clear? Or maybe catch that case specifically and print a critical message and `fail` properly?

Reply:
Done, if `reply()` is empty and `status != SyncFileItem::FileIgnored` a graceful fail is implemented with a `qCCritical` message.

Reply:
Yes, primarily it's for signalling to readers that this is not indicative of a bug. a commented `ENFORCE(job->reply());` would have done it as well.